pkg/autoscaler/autoscaler/pdplan_autoscaler.go

```diff
@@ -128,6 +131,108 @@ func (am *autoScalerManager) updateAutoscalingClusters(tac *v1alpha1.TidbCluster
-	// TODO in next PR
-	return nil
+	var errs []error
+	for _, group := range groupsToCreate {
+		plan := groupPlanMap[group]
+		component := plan.Component
+
+		if !checkAutoscalingComponent(tac, component) {
+			continue
+		}
+
+		resource, ok := tac.Spec.Resources[plan.ResourceType]
+		if !ok {
+			errs = append(errs, fmt.Errorf("unknown resource type %v for group %s tac[%s/%s]", plan.ResourceType, plan.Labels[groupLabelKey], tac.Namespace, tac.Name))
+			continue
+		}
+		resList := corev1.ResourceList{
+			corev1.ResourceCPU:    resource.CPU,
+			corev1.ResourceMemory: resource.Memory,
+		}
+
+		autoTcName, err := genAutoClusterName(tac, component, plan.Labels, resource)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		autoTc := &v1alpha1.TidbCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      autoTcName,
+				Namespace: tc.Namespace,
+				OwnerReferences: []metav1.OwnerReference{
+					controller.GetTiDBClusterAutoscalerOwnerRef(tac),
+				},
+			},
+			Status: v1alpha1.TidbClusterStatus{
+				AutoScaler: &v1alpha1.TidbClusterAutoScalerRef{
+					Name:      tac.Name,
+					Namespace: tac.Namespace,
+				},
+			},
+			Spec: v1alpha1.TidbClusterSpec{
+				Cluster: &v1alpha1.TidbClusterRef{
+					Name:      tc.Name,
+					Namespace: tc.Namespace,
+				},
+			},
+		}
+
+		switch component {
+		case v1alpha1.TiKVMemberType.String():
+			resList[corev1.ResourceStorage] = resource.Storage
+
+			autoTc.Spec.TiKV = tc.Spec.TiKV.DeepCopy()
+			autoTc.Spec.TiKV.Replicas = int32(plan.Count)
+			autoTc.Spec.TiKV.ResourceRequirements = corev1.ResourceRequirements{
+				Limits:   resList,
+				Requests: resList,
+			}
+			if autoTc.Spec.TiKV.Config == nil {
+				autoTc.Spec.TiKV.Config = &v1alpha1.TiKVConfig{
+					Server: &v1alpha1.TiKVServerConfig{
+						Labels: map[string]string{},
+					},
+				}
+			}
+			if autoTc.Spec.TiKV.Config.Server == nil {
+				autoTc.Spec.TiKV.Config.Server = &v1alpha1.TiKVServerConfig{
+					Labels: map[string]string{},
+				}
+			}
+			if autoTc.Spec.TiKV.Config.Server.Labels == nil {
+				autoTc.Spec.TiKV.Config.Server.Labels = map[string]string{}
+			}
+			for k, v := range plan.Labels {
+				autoTc.Spec.TiKV.Config.Server.Labels[k] = v
+			}
+		case v1alpha1.TiDBMemberType.String():
+			autoTc.Spec.TiDB = tc.Spec.TiDB.DeepCopy()
+			autoTc.Spec.TiDB.ResourceRequirements = corev1.ResourceRequirements{
+				Limits:   resList,
+				Requests: resList,
+			}
+			if autoTc.Spec.TiDB.Config == nil {
+				autoTc.Spec.TiDB.Config = &v1alpha1.TiDBConfig{
+					Labels: map[string]string{},
+				}
+			}
+			if autoTc.Spec.TiDB.Config.Labels == nil {
+				autoTc.Spec.TiDB.Config.Labels = map[string]string{}
+			}
+			for k, v := range plan.Labels {
+				autoTc.Spec.TiDB.Config.Labels[k] = v
+			}
+		}
+
+		// Patch custom labels
+		patchAutoscalingLabels(autoTc, tac, component, group)
```

Comment:
Why not directly set the label?

Reply:
Updated to set the labels while creating the new TidbCluster object